xc-admin/packages/xc-admin-common/src/multisig.ts

```diff
@@ -0,0 +1,80 @@
+import { PublicKey } from "@solana/web3.js";
+import Squads, {
+  DEFAULT_MULTISIG_PROGRAM_ID,
+  getIxPDA,
+  getTxPDA,
+} from "@sqds/mesh";
+import { InstructionAccount, TransactionAccount } from "@sqds/mesh/lib/types";
+import BN from "bn.js";
+import lodash from "lodash";
+
+export async function getActiveProposals(
+  squad: Squads,
+  vault: PublicKey,
+  offset: number = 1
+): Promise<TransactionAccount[]> {
+  const msAccount = await squad.getMultisig(vault);
+  let txKeys = lodash
+    .range(offset, msAccount.transactionIndex + 1)
+    .map((i) => getTxPDA(vault, new BN(i), DEFAULT_MULTISIG_PROGRAM_ID)[0]);
+  let msTransactions = await squad.getTransactions(txKeys);
+  return msTransactions
+    .filter(
+      (x: TransactionAccount | null): x is TransactionAccount => x != null
+    )
+    .filter((x) => lodash.isEqual(x.status, { active: {} }));
+}
+
+export async function getManyProposalsInstructions(
+  squad: Squads,
+  txAccounts: TransactionAccount[]
+): Promise<InstructionAccount[][]> {
+  let allIxsKeys = [];
+  let ownerTransaction = [];
+  for (let [index, txAccount] of txAccounts.entries()) {
```

Comment:
seems simpler to reuse `getProposalInstructions` here?

i think this whole function is basically `txAccounts.map(a => getProposalInstructions(squads, a)`?

Reply:
Nah to goal of `getManyProposalsInstructions` is requesting all the instruction accounts for all proposals in one single RPC call.
`txAccounts.map(a => getProposalInstructions(squads, a)` would make 1 RPC call per proposal.